### PR TITLE
Clean data mask ourselves

### PR DESCRIPTION
**Pull request by lionel-**
Part of #382

### CODE REVIEW

File: tests/testthat/test-hybrid.R [hunk 1]
```diff
@@ -7,9 +7,9 @@ test_that("hybrid evaluation environment is cleaned up (#2358)", {
   df <- mutate(df, g = list(quo(.)))
   df <- mutate(df, h = list(~.))
 
-  expect_environments_clean(df$f[[1]])
```
Comment (krlmlr) [inline tests/testthat/test-hybrid.R]: I don't understand the effect of this change. What do the environments of `f[[1]]` etc. contain?

Comment (lionel-) [reply tests/testthat/test-hybrid.R]: Objects installed by `rlang::new_data_mask()`: a tidyeval flag and the `~` binding.

Comment (krlmlr) [reply tests/testthat/test-hybrid.R]: Can we at least check that these environments don't contain any variables (`a:h` in our case)?

Comment (lionel-) [reply tests/testthat/test-hybrid.R]: I don't think we need to do that, the mask bottom is private to rlang.

Comment (lionel-) [reply tests/testthat/test-hybrid.R]: I'll add an `env_has(inherit = TRUE)` check just to be sure.

**Pull request merged**